backends-velox/src/main/scala/org/apache/spark/sql/execution/VeloxColumnarWriteFilesExec.scala

```diff
@@ -73,6 +81,133 @@ case class VeloxWriteFilesMetrics(
+      val metrics = objectMapper
+        .readValue(fragments.toString.getBytes("UTF-8"), classOf[VeloxWriteFilesMetrics])
+      logDebug(s"Velox write files metrics: $metrics")
+
+      val fileWriteInfos = metrics.fileWriteInfos
+      assert(fileWriteInfos.length == 1)
+      val fileWriteInfo = fileWriteInfos.head
+      numBytes += fileWriteInfo.fileSize
+      val targetFileName = fileWriteInfo.targetFileName
+      val outputPath = writeFilesSpec.description.path
+
+      // part1=1/part2=1
+      val partitionFragment = metrics.name
+      // write a non-partitioned table
+      if (partitionFragment != "") {
+        updatedPartitions += partitionFragment
+        val tmpOutputPath = outputPath + "/" + partitionFragment + "/" + targetFileName
+        val customOutputPath = writeFilesSpec.description.customPartitionLocations.get(
+          PartitioningUtils.parsePathFragment(partitionFragment))
+        if (customOutputPath.isDefined) {
+          addedAbsPathFiles(tmpOutputPath) = customOutputPath.get + "/" + targetFileName
+        }
+      }
+    }
+
+    // Reports bytesWritten and recordsWritten to the Spark output metrics.
+    Option(TaskContext.get()).map(_.taskMetrics().outputMetrics).foreach {
+      outputMetrics =>
+        outputMetrics.setBytesWritten(numBytes)
+        outputMetrics.setRecordsWritten(numWrittenRows)
+    }
+
+    val partitionsInternalRows = updatedPartitions.map {
+      part =>
+        val parts = new Array[Any](1)
+        parts(0) = part
+        new GenericInternalRow(parts)
+    }.toSeq
+    val stats = BasicWriteTaskStats(
+      partitions = partitionsInternalRows,
+      numFiles = loadedCb.numRows() - 1,
+      numBytes = numBytes,
+      numRows = numWrittenRows)
+    val summary =
+      ExecutedWriteSummary(updatedPartitions = updatedPartitions, stats = Seq(stats))
+
+    WriteTaskResult(new TaskCommitMessage(addedAbsPathFiles.toMap -> updatedPartitions), summary)
+  }
+
+  override def compute(split: Partition, context: TaskContext): Iterator[WriterCommitMessage] = {
+    val commitProtocol = new SparkWriteFilesCommitProtocol(
+      jobTrackerID,
+      writeFilesSpec.description,
+      writeFilesSpec.committer)
+
+    commitProtocol.setupTask()
+    val writePath = commitProtocol.newTaskAttemptTempPath()
+    logDebug(s"Velox staging write path: $writePath")
+    var resultColumnarBatch: ColumnarBatch = null
+    try {
+      Utils.tryWithSafeFinallyAndFailureCallbacks(block = {
+        BackendsApiManager.getIteratorApiInstance.injectWriteFilesTempPath(writePath)
+
+        // initialize the native plan
+        val iter = firstParent[ColumnarBatch].iterator(split, context)
+        assert(iter.hasNext)
+        resultColumnarBatch = iter.next()
+        commitProtocol.commitTask()
+      })(
+        catchBlock = {
+          // If there is an error, abort the task
+          commitProtocol.abortTask()
+          logError(s"Job ${commitProtocol.getJobId} aborted.")
```

Comment:
Do we need to log `taskID` here?

Reply:
it is copied from Spark code, the logging should contain the task id